src/main/java/seedu/address/model/team/Team.java

```diff
@@ -3,75 +3,132 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
 
+import java.util.ArrayList;
 import java.util.List;
 
 import seedu.address.model.contact.Contact;
 import seedu.address.model.contact.Name;
 import seedu.address.model.contact.UniqueContactList;
 
-/**
- * Represents a team in CodeConnect.
- * Guarantees: details are present and not null, field values are validated, immutable.
- */
 public class Team {
     private final Name name;
     private final UniqueContactList members = new UniqueContactList();
 
+  /**
+   * Constructs a {@code Team} with the given name.
+   *
+   * @param name A valid name for the team.
+   */
+  public Team(Name name) {
+      requireNonNull(name);
+      this.name = name;
+  }
+
+  /**
+   * Additional constructor for creating a team with members.
+   *
+   * @param name    A valid name for the team.
+   * @param members A list of contacts to be members of the team.
+   */
+  public Team(Name name, List<Contact> members) {
+      requireAllNonNull(name, members);
+      this.name = name;
+      this.members.setContacts(members);
+  }
+
     /**
-     * Every field must be present and not null.
+     * Returns the name of the team.
+     *
+     * @return The name of the team.
      */
-    public Team(Name name) {
-        requireNonNull(name);
-        this.name = name;
+    public Name getName() {
+        return name;
     }
 
     /**
-     * Additional constructor for creating a team with members.
+     * Returns an unmodifiable list of members in the team.
+     *
+     * @return An unmodifiable list of members in the team.
      */
-    public Team(Name name, List<Contact> members) {
-        requireAllNonNull(name, members);
-        this.name = name;
-        this.members.setContacts(members);
+    public List<Contact> getMembers() {
+        return members.asUnmodifiableObservableList();
     }
 
-    public Name getName() {
-        return name;
+    /**
+     * Checks if the given contact is a member of the team.
+     *
+     * @param contact The contact to check.
+     * @return True if the contact is a member of the team, false otherwise.
+     */
+    public boolean hasMember(Contact contact) {
+        return members.contains(contact);
     }
 
-    public List<Contact> getMembers() {
-        return members.asUnmodifiableObservableList();
+    /**
+     * Adds a member to the team.
+     * Sorts the members alphabetically after adding.
+     *
+     * @param contact The contact to add as a member.
+     */
+    public void addMember(Contact contact) {
+        requireNonNull(contact);
+        members.add(contact);
+        sortMembersAlphabetically();
+    }
```

Comment:
I think its better if `Team` is immutable/read-only similar to `Contact`, so you should probably remove these methods.